src/cluster_asm.c

```diff
@@ -2804,6 +2804,18 @@ void trimslotsCommand(client *c) {
          * trimmed. We have to trim the keys synchronously. */
         clusterDelKeysInSlotRangeArray(slots, 1);
     } else {
+        /* We can not trim the slot that this node is serving. */
+        if (clusterNodeIsMaster(getMyClusterNode())) {
+            for (int i = 0; i < slots->num_ranges; i++) {
+                for (int j = slots->ranges[i].start; j <= slots->ranges[i].end; j++) {
+                    if (clusterCanAccessKeysInSlot(j)) {
+                        addReplyErrorFormat(c, "the slot %d is served by this node", j);
+                        slotRangeArrayFree(slots);
+                        return;
+                    }
+                }
+            }
+        }
```

Comment:
I was also thinking if we can should `asmTrimJobSchedule()` instead of `asmTrimSlots()` to handle it in beforeSleep(). Otherwise, it will crash if the server is paused. Maybe you can make that change as well.

Reply:
maybe I'm confused. If writes are paused, user can't even execute this command. So, perhaps there will be no problem.
 
Previously, we suffered from assert/crash because we were trying to propagate trimslot during write pause. So, we moved handling it to beforeSleep(). Though, this seems unrelated to this scenario.